graphql-kotlin-schema-generator/src/main/kotlin/com/expedia/graphql/generator/state/TypesCache.kt

```diff
@@ -35,12 +35,12 @@ internal class TypesCache(private val supportedPackages: List<String>) {
         return null
     }
 
-    fun put(key: TypesCacheKey, kGraphQLType: KGraphQLType): KGraphQLType? {
+    fun saveIntoCache(key: TypesCacheKey, kGraphQLType: KGraphQLType): KGraphQLType? {
         val cacheKey = getCacheKeyString(key)
 
         if (cacheKey != null) {
-            typeUnderConstruction.remove(kGraphQLType.kClass)
-            return cache.put(cacheKey, kGraphQLType)
+            cache[cacheKey] = kGraphQLType
```

Comment:
We don't need to remove from `typeUnderConstruction` since the only place we add is in the `buildIfNotUnderConstruction` method and we also remove it there.